src/restconfig/src/test/java/org/geoserver/rest/resources/ResourceTest.java

```diff
@@ -0,0 +1,310 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.rest.resources;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+import net.sf.json.JSON;
+import net.sf.json.JSONObject;
+import net.sf.json.test.JSONAssert;
+
+import org.custommonkey.xmlunit.NamespaceContext;
+import org.custommonkey.xmlunit.SimpleNamespaceContext;
+import org.custommonkey.xmlunit.XMLAssert;
+import org.custommonkey.xmlunit.XMLUnit;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
+import org.geoserver.rest.util.IOUtils;
+import org.geoserver.test.GeoServerSystemTestSupport;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.w3c.dom.Document;
+
+import com.mockrunner.mock.web.MockHttpServletResponse;
+
+/**
+ * 
+ * @author Niels Charlier
+ *
+ */
+public class ResourceTest extends GeoServerSystemTestSupport {
```

Comment:
Besides, this test also detects from file extension (but not from content, the above does both): 
https://github.com/geoserver/geoserver/pull/1487/files#diff-53d3ea4d2b8d4e1363c0415f421cdda9R133